### PR TITLE
feat: suggest `deno install --entrypoint` instead of `deno cache`

### PR DESCRIPTION
**Pull request by iuioiua**
Hides `deno cache` from `--help` output.

### CODE REVIEW

File: cli/args/flags.rs [hunk 1]
```diff
@@ -4165,6 +4170,7 @@ fn bundle_parse(flags: &mut Flags, matches: &mut ArgMatches) {
 }
 
 fn cache_parse(flags: &mut Flags, matches: &mut ArgMatches) {
+  log::info!("{}", colors::yellow("⚠️ Warning: `deno cache` is deprecated and will be removed in Deno 2.1.\nUse `deno install` instead."));
```
Comment (bartlomieju) [inline cli/args/flags.rs:4173]: This has side-effects for flag parsing - which is an anti-pattern. Can you move this warning to the `main.rs` file where `deno cache` is actually invoked?

**Comment by marvinhagemeister** [comment]
Note: We probably should point to the planned `--entrypoint` flag for the install command in those messages added in this PR once that lands.

**Review by bartlomieju — CHANGES_REQUESTED** [review]
Waiting on `deno install -e` support before landing this one, which @nathanwhit is working on.

**Comment by KnorpelSenf** [comment]
I don't really understand why `--entrypoint` tells me “this thing now doesn't install anything but it populates the cache instead.” Seems like very poor naming to me. What am I missing?

Also, why do you think it's a great idea to overload `deno install` with more and more features? It recently proliferated from doing 1 thing and doing it well to now doing 3 things in less obvious ways, depending on the flags you pass. 

I do agree that it's useful to have a single command that both adds and caches a module, but if you break everybody's workflow anyway, why not end up with less ambiguity?

**Comment by greentore** [comment]
Yeah, I don't get this change at all. The new way is both less ergonomic and less descriptive. `install --entrypoint`  tells me absolutely nothing.

**Comment by bartlomieju** [comment]
FYI we are still discussing this internally if we should proceed. We are hearing your feedback.

**Comment by bartlomieju** [comment]
The decision has been made to keep `deno cache` as is - but it will be hidden from the help output and the error messages will suggest using `deno install --entrypoint`. In other words - don't worry, your existing workflows will keep working as before.

**Comment by uasi** [comment]
> The decision has been made to keep deno cache as is

Glad to hear that. Still don't get what `--entrypoint` means - why not `--cache-only` or  `--deps` or something?

**Review by bartlomieju — APPROVED** [review]
LGTM

**Comment by KnorpelSenf** [comment]
Is `deno cache mod.ts` going to be exactly equivalent to `deno install --entrypoint mod.ts`? Like an alias? I'm still trying to understand what `deno install` is doing. My current understanding is that
- it can modify the import map in `deno.json` (but not a separate import map file) in order to provide shorter imports,
- it can create a very short bash script file that makes a Deno application available in the `$PATH` of the machine, and
- it can cache modules 

and sometimes it does various of these things at once. Is that correct? The docs currently only mention the second thing.

**Comment by bartlomieju** [comment]
> Is deno cache mod.ts going to be exactly equivalent to deno install --entrypoint mod.ts? Like an alias?

Yes

> it can modify the import map in deno.json (but not a separate import map file) in order to provide shorter imports,

Correct

> it can create a very short bash script file that makes a Deno application available in the $PATH of the machine, and

Yes, when you use `-g` flag

> it can cache modules

Pretty much, which is the same as the first point, but instead of relying on `deno.json` you point it to a file that has the dependencies listed (like a single-file script).

> and sometimes it does various of these things at once. Is that correct? 

Not really, it's one of the three modes depending if you run `deno install`, `deno install -g` or `deno install -e`.

> The docs currently only mention the second thing.

We are currently updating the docs

**Comment by nathanwhit** [comment]
As a mental model, you can think of `deno install` as the command for installing dependencies. 

So `deno install --global <dep>` installs a dependency globally, `deno install` (no arg) installs local dependencies of your project (ones listed in package.json, deno.json, or import map), `deno install pkg` installs a new dependency to your project (adding it to your config). `deno install --entrypoint <file>` is installing dependencies required by a file, treating that file as the entrypoint.

**Comment by uasi** [comment]
> `deno install --entrypoint <file>` is installing dependencies required by a file, treating that file as the entrypoint.

That clears up my question, thank you.

**Pull request merged**